src/drivers/tattu_can/TattuCan.cpp

```diff
@@ -0,0 +1,221 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2018-20 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "TattuCan.hpp"
+
+#include <drivers/drv_hrt.h>
+#include <systemlib/mavlink_log.h>
+#include <px4_platform_common/shutdown.h>
+#include <px4_platform_common/getopt.h>
+#include <px4_platform_common/log.h>
+
+extern orb_advert_t mavlink_log_pub;
+
+TattuCan::TattuCan() :
+	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::tattucan)
+{
+}
+
+TattuCan::~TattuCan()
+{
+}
+
+void TattuCan::Run()
+{
+	if (should_exit()) {
+		exit_and_cleanup();
+		return;
+	}
+
+	if (!_initialized) {
+
+		_fd = ::open("/dev/can0", O_RDWR);
```

Comment:
Shorter term we could make it easy to have UAVCAN optionally only on CAN1 and run this on CAN2.

Reply:
I noticed that you have to start this driver AFTER uavcan driver starts, otherwise uavcan steals the bus and it doesn't work (Even with UAVCAN_INTERFACES set to 1)

Reply:
At the moment this should not be allowed to run at the same time as UAVCAN at all.

Reply:
I tested it on v1.11.0 starting it after uavcan and it works alongside an ARK GPS on CAN1.